targets/TARGET_STM/TARGET_STM32L4/TARGET_STM32L475xG/TARGET_DISCO_L475VG_IOT01A/PeripheralPinMaps.h

```diff
@@ -52,6 +52,94 @@
 #include "PinNamesTypes.h"
 #include <mstd_cstddef>
 
+//*** GPIO ***
+// Pinmap used for FPGA GPIO testing only
+MSTD_CONSTEXPR_OBJ_11 PinMap PinMap_GPIO[] = {
```

Comment:
I'm still not a big fan of this proposal.
espacially this is really are to read - this is supposed to be an exclusion table and it actually contains all the defined pins with dummy values ... can't we at least create an enum the list of pins that need to be excluded from the test ? 
That would be easier to read and create  .

Reply:
This is done in the same way as other peripherals. Please check PWM pin-map table where also some pins are commented out because of some reason:

https://github.com/ARMmbed/mbed-os/blob/98af2d0b375c4b7d2c4c823e4340cc8c3b05875c/targets/TARGET_STM/TARGET_STM32L4/TARGET_STM32L475xG/TARGET_DISCO_L475VG_IOT01A/PeripheralPinMaps.h#L257-L260

We have a ready testing utility functions which can deal with the pin-maps. So from our point of view making exceptions here is very problematic.

Reply:
Except thePWM and pther  tables are maps between pins and HW block instances ///
The GPIO one is a table mapping pins to nothing ...  :-(

Reply:
What you actually need is a simple list of pins on which you don't want to run a GPIO test. Do you need a table because other MCUs from other vendors a GPIO instance value ?

Reply:
After a quick chat with @maciejbocianski @mprse , we think the better way of the achieve the same result as @LMESTM suggested is add `const PinList *pinmap_restricted_gpio_pins()` which exclude the GPIO pins can't be tested, this PR will be updated along with some other targets also need to be updated as well

Reply:
Here I added a mechanism to handle list of restricted GPIO in FPGA testing:
https://github.com/ARMmbed/mbed-os/pull/12379#issuecomment-583313474

Also here I dropped base commit and added commit with the list of restricted GPIO pins instead.